### PR TITLE
Aos 2 ric remove edges efif

### LINKED ISSUE #1967: Arr_trapezoid_ric_point_location: Strange warning when adding/removing edges
## Issue Details

When a ```Arr_trapezoid_ric_point_location``` is attached to an ```Arrangement_2```, after an isolated inner CCB is added to the arrangement, and then the edges are removed one by one, the locator will raise a strange warning, although nothing wrong was done on the arrangement.

## Source Code

```C++

#include <CGAL/Cartesian.h>
#include <CGAL/Arr_segment_traits_2.h>
#include <CGAL/Arrangement_2.h>
#include <CGAL/Arr_walk_along_line_point_location.h>
#include <CGAL/Arr_trapezoid_ric_point_location.h>

typedef CGAL::Exact_predicates_exact_constructions_kernel Kernel;
typedef Kernel::FT                                        Number_type;
typedef CGAL::Arr_segment_traits_2<Kernel>                Traits;
typedef Traits::Point_2                                   Point;
typedef Traits::X_monotone_curve_2                        Segment;
typedef Traits::Curve_2                                   Curve;
typedef CGAL::Arrangement_2<Traits>                       Arrangement;

typedef CGAL::Arr_trapezoid_ric_point_location<Arrangement>   TrapezoidLocator;

int main (int /*argv**/, char ** /*argv*/)
{
    Arrangement arrangement;
    TrapezoidLocator locator(arrangement);

    // First create a new face
    auto v1 = arrangement.insert_in_face_interior(Point(0, 0), arrangement.unbounded_face());
    auto v2 = arrangement.insert_in_face_interior(Point(0, 1), arrangement.unbounded_face());
    auto v3 = arrangement.insert_in_face_interior(Point(1, 1), arrangement.unbounded_face());
    auto v4 = arrangement.insert_in_face_interior(Point(1, 0), arrangement.unbounded_face());
    auto e1 = arrangement.insert_at_vertices(Segment(v1->point(), v2->point()), v1, v2);
    auto e2 = arrangement.insert_at_vertices(Segment(v2->point(), v3->point()), v2, v3);
    auto e3 = arrangement.insert_at_vertices(Segment(v3->point(), v4->point()), v3, v4);
    auto e4 = arrangement.insert_at_vertices(Segment(v4->point(), v1->point()), v4, v1);

    // To trigger the warning, we need to add an extra inner CCB in the unbounded face ...
    auto v5 = arrangement.insert_in_face_interior(Point(2, 2), arrangement.unbounded_face());
    auto v6 = arrangement.insert_in_face_interior(Point(3, 3), arrangement.unbounded_face());
    auto e5 = arrangement.insert_at_vertices(Segment(v5->point(), v6->point()), v5, v6);

    // ... then start removing the edges
    arrangement.remove_edge(e5);
    arrangement.remove_edge(e4);
    arrangement.remove_edge(e3);
    arrangement.remove_edge(e2);
    arrangement.remove_edge(e1); // This one will generate a warning

    return 0;
}
```
Which gives:
```
CGAL warning: check violation!
Expression : lt1==POINT && lt2==POINT
File       : /usr/include/CGAL/Arr_point_location/Trapezoidal_decomposition_2_impl.h
Line       : 1781
Explanation: 
Refer to the bug-reporting instructions at http://www.cgal.org/bug_report.html
```

## Environment

KUbuntu 16.10, with distro provided CGAL-4.8.1, boost-1.61.0.2 and gcc-6.1.1.

### PR DESCRIPTION
**Pull request by efifogel**
This pull requests fixes a few bugs in the code of the trapezoidal RIC point location of the 2D Arrangement package that handles removal of halfedges from the arrangement. One of the fixed bugs has already been reported and is listed in the tracker.

The pull requests does the following:
1. It fixes issues 1967.
2. It fixes a couple of additional bugs in the trapezoidal RIC point location that handles removals of halfedges from the arrangement.
3. It introduces 2 new tests into the test suit, which covers the fixes above.
4. It rearranged the point-location test to enable convenient enabling and disabling of tested point-location strategies.

## Release Management

* Affected package(s): Arrangement_on_surface_2
* Issue(s) solved (if any): fix #1967 
* Feature/Small Feature (if any): None

### CODE REVIEW

File: Arrangement_on_surface_2/test/Arrangement_on_surface_2/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,17 @@ project( Arrangement_on_surface_2_Tests )
...
 enable_testing()
 
-cmake_minimum_required(VERSION 2.8.10)
+cmake_minimum_required(VERSION 3.1)
+
+list(FIND CMAKE_CXX_COMPILE_FEATURES cxx_generalized_initializers has_cpp11)
+if (has_cpp11 LESS 0)
+  message(STATUS "NOTICE: These examples requires a C++11 compiler and will not be compiled.")
+  return()
+endif()
```
Comment (lrineau) [inline Arrangement_on_surface_2/test/Arrangement_on_surface_2/CMakeLists.txt]: I find that strange. Does anything in the code of the Arr_2 use C++11 code?

**Comment by lrineau** [comment]
TODO: I have flagged that PR with "TODO" because I want to avoid merging it in `master` before the question/issue about C++11 (https://github.com/CGAL/cgal/pull/3093#pullrequestreview-120939283) is solved.

**Comment by efifogel** [comment]
No, the code itself and the code of the examples and demo does not use
C++11, only the code of the tests does.


   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by lrineau** [comment]
That should be fixed. Otherwise, `Arrangement_on_surface_2` will not be tested on pre-C++11 platforms. That is more or less forbidden by our current rules: only new packages or new features can be C++11-only. Or demos or specific examples.

**Comment by efifogel** [comment]
I'll fix it.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by efifogel** [comment]
It's fixed now.
I've suppress C++11 for now.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

File: Minkowski_sum_2/test/Minkowski_sum_2/CMakeLists.txt [hunk 2]
```diff
@@ -6,6 +6,16 @@ project( Minkowski_sum_2_Tests )
...
+list(FIND CMAKE_CXX_COMPILE_FEATURES cxx_generalized_initializers has_cpp11)
+if (has_cpp11 LESS 0)
+  message(STATUS "NOTICE: These examples requires a C++11 compiler and will not be compiled.")
+  return()
+endif()
+
+# Use C++11 for this directory and its sub-directories.
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+
```
Comment (lrineau) [inline Minkowski_sum_2/test/Minkowski_sum_2/CMakeLists.txt]: Actually, by "fix" I meant "the testsuite of Minkowski_sum_2" should not use C++11! :-\

**Comment by efifogel** [comment]
I see.
As much as I hated to do it, I guess you were right, so I did it.


   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by lrineau** [comment]
Successfully tested in [CGAL-4.13-Ic-21][*], and then merged.

  [*]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.13-Ic-21.shtml

**Pull request merged**